dev/com.ibm.ws.security.wim.adapter.ldap_fat/fat/src/com/ibm/ws/security/wim/adapter/ldap/fat/LoginPropertyTest.java

```diff
@@ -0,0 +1,290 @@
+     * Log in the with the user's cn from userFilter. No loginProperties.
+     *
+     * @throws Exception If there was an unexpected exception.
+     */
+    @Test
+    public void testNoLoginPropertyWithFilter() throws Exception {
+        updateLibertyServer(true);
+        assertDNsEqual("Authentication should succeed.",
+                       USER_DN, servlet.checkPassword(userCn, "password"));
+    }
+
+    /**
+     * Log in the with the user's sn from loginProperties. Include userFilter.
+     * This test should issue a warning that the userFilter is ignored and
+     * the dynamically built filter is used.
+     * Additionally, we expect the user display name to be sn because loginProperty
+     * takse precedence.
+     *
+     * @throws Exception If there was an unexpected exception.
+     */
+    @Test
+    public void testLoginPropertyWithFilter() throws Exception {
+        updateLibertyServer(true, "sn");
+        assertDNsEqual("Authentication should succeed.",
+                       USER_DN, servlet.checkPassword(userSn, "password"));
+        assertNotNull("Should find CWIML4505W message.", libertyServer.waitForStringInLog("CWIML4505W"));
+        assertEquals(userSn, servlet.getUserDisplayName(USER_DN));
+    }
+
+    /**
+     * Log in the with the user's sn from loginProperties.
+     * This test uses the dynamically created user filter.
+     *
+     * @throws Exception If there was an unexpected exception.
+     */
+    @Test
+    public void testLoginProperty() throws Exception {
+        updateLibertyServer(false, "sn");
+        assertDNsEqual("Authentication should succeed.",
+                       USER_DN, servlet.checkPassword(userSn, "password"));
+    }
+
+    /**
+     * Log in the with the user's sn and uid from loginProperties.
+     * This test uses the dynamically created user filter.
+     * Logging in with uid should fail.
+     * Additionally, we expect the user display name to be cn as this is
+     * the first listed loginProperty.
+     *
+     * @throws Exception If there was an unexpected exception.
+     */
+    @Test
+    public void testMultipleLoginProperty() throws Exception {
+        updateLibertyServer(false, "cn", "sn");
+        assertDNsEqual("Authentication should succeed.",
+                       USER_DN, servlet.checkPassword(userSn, "password"));
+        assertDNsEqual("Authentication should succeed.",
+                       USER_DN, servlet.checkPassword(userCn, "password"));
+        assertNull("Authentication should not succeed.", servlet.checkPassword(userUid, "password"));
+        assertEquals(userCn, servlet.getUserDisplayName(USER_DN));
+    }
+
+    /**
+     * Attempt login the with the user's uid. The loginProperty
+     * defined is sn so this test should fail.
+     * This test uses the dynamically created user filter.
+     *
+     * @throws Exception If there was an unexpected exception.
+     */
+    @Test
+    public void testNonLoginProperty() throws Exception {
```

Comment:
I think the negative test should be done for each attribute in all the other tests. SO we test login for UID, SN and CN in each test verifying that the positive and negative tests pass. Then you can remove this one.